workbench/_api/routes/forward_pass.py

```
@@ -0,0 +1,101 @@
+            f"{state.ndif_backend_url}/response/{job_id}",
+            timeout=10,
+        )
+        if resp.status_code != 200:
+            raise HTTPException(status_code=502, detail=f"NDIF status returned {resp.status_code}")
+        payload = resp.json()
+        status_value = payload.get("status", "UNKNOWN")
+        description = payload.get("description") or payload.get("msg")
+        return {"status": status_value, "description": description}
+    except requests.RequestException as exc:
+        logger.warning(f"forward_pass status proxy failed for {job_id}: {exc}")
+        raise HTTPException(status_code=502, detail="NDIF status unreachable")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)